### PR TITLE
RavenDB-5438 Implemented a persistent cache for database info (to be …

### PR DESCRIPTION
**Pull request by talweiss1982**
…displayed in the studio for unloaded databases)

### CODE REVIEW

File: src/Raven.Server/Documents/DatabaseInfoCache.cs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+        public unsafe void InsertDatabaseInfo(DynamicJsonValue databaseInfo)
+        {
+            TransactionOperationContext context;
+            using (_contextPool.AllocateOperationContext(out context))
+            using (var tx = context.OpenWriteTransaction())
+            {
+                var table = tx.InnerTransaction.OpenTable(_databaseInfoSchema, DatabaseInfoSchema.DatabaseInfoTree);
+
+                var databaseName = databaseInfo["Name"] as string;
```
Comment (ayende) [inline src/Raven.Server/Documents/DatabaseInfoCache.cs]: Promote to parameter externally

File: src/Raven.Server/Documents/DocumentDatabase.cs [hunk 2]
```diff
@@ -290,6 +299,34 @@ public void Dispose()
             exceptionAggregator.ThrowIfNeeded();
         }
 
+        private static readonly string CachedDatabaseInfo = "CachedDatabaseInfo";
+        public DynamicJsonValue GenerateDatabaseInfo(bool forCache)
```
Comment (ayende) [inline src/Raven.Server/Documents/DocumentDatabase.cs]: Removed `forCache` if not used, YAGNI for now

File: src/Raven.Server/Documents/DatabasesLandlord.cs [hunk 3]
```diff
@@ -129,6 +130,12 @@ private DocumentDatabase CreateDocumentsStorage(StringSegment databaseName, Rave
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void UpdateDatabaseInfoCache(DocumentDatabase database, ServerStore serverStore)
+        {
+            serverStore.DatabaseInfoCache.InsertDatabaseInfo(database.GenerateDatabaseInfo(true));
```
Comment (ayende) [inline src/Raven.Server/Documents/DatabasesLandlord.cs]: Here delete this on load, so we'll only have this for dbs that were successfully shutdown.

**Review by ayende — CHANGES_REQUESTED** [review]

**Comment by talweiss1982** [comment]
something went wrong with the rebase

**Pull request closed without merging**